packages/editor/src/bindings/post-meta.js

```diff
@@ -16,13 +16,64 @@ export default {
 		return args.key;
 	},
 	getValue( { registry, context, args } ) {
+		const postId = context.postId
+			? context.postId
+			: registry.select( editorStore ).getCurrentPostId();
 		const postType = context.postType
 			? context.postType
 			: registry.select( editorStore ).getCurrentPostType();
 
 		return registry
 			.select( coreDataStore )
-			.getEditedEntityRecord( 'postType', postType, context.postId )
-			.meta?.[ args.key ];
+			.getEditedEntityRecord( 'postType', postType, postId ).meta?.[
+			args.key
+		];
+	},
+	setValue( { registry, context, args, value } ) {
+		const postId = context.postId
+			? context.postId
+			: registry.select( editorStore ).getCurrentPostId();
+		const postType = context.postType
+			? context.postType
+			: registry.select( editorStore ).getCurrentPostType();
+		registry
+			.dispatch( coreDataStore )
+			.editEntityRecord( 'postType', postType, postId, {
+				meta: {
+					[ args.key ]: value,
+				},
+			} );
+	},
+	lockAttributesEditing( { select, context = {}, args } ) {
+		const postId = context.postId
+			? context.postId
+			: select( editorStore ).getCurrentPostId();
+		const postType = context.postType
+			? context.postType
+			: select( editorStore ).getCurrentPostType();
+
+		// Check that editing is happening in the post editor and not a template.
+		if ( postType === 'wp_template' ) {
+			return true;
+		}
+
+		// Check that the custom field is not protected and available in the REST API.
+		const isFieldExposed =
+			select( editorStore ).getEditedPostAttribute( 'meta' )[ args.key ];
+		if ( ! isFieldExposed ) {
+			return true;
+		}
+
+		// Check that the user has the capability to edit post meta.
+		const canUserEdit = select( coreDataStore ).canUserEditEntityRecord(
+			'postType',
+			postType,
+			postId
+		);
+		if ( ! canUserEdit ) {
+			return true;
+		}
+
+		return false;
```

Comment:
I've been thinking about this and, if your are a contributor or an author user. You are not even allowed to access to the post editor part. So does it make sense to have this check? 

I think it will be always true (this is only called in the post editor page, meaning that the user is an author/contributor of their own posts, an editor or an admin, canUserEditEntityRecord is true in all of those cases)

![Screenshot 2024-05-17 at 16 57 44](https://github.com/WordPress/gutenberg/assets/37012961/81d32162-2d6a-4187-b7cf-27823c674263)

Reply:
> So does it make sense to have this check?

I was wondering the same, but I left it just in case. @gziolo Any idea if this is necessary at all?

Reply:
What if you are editing a post as an author and there is a custom block that sets the post id in the context to a different value in all inner blocks. That potentially could be also a Query Loop that has some core blocks with connected attributes. As an author you shouldn’t be able to edit anything inside these nested connected block in the posts you don’t authored. It’s just a single example but options  are endless with custom blocks. Another example would be a feature product block that allows customizations in inner blocks. A different set of challenges will start to arise with more sources. How do you prevent editing the site title, site logo or site tagline when injected in the post content?

Reply:
Thanks for the examples. I think they totally make sense. It seems it is worth keeping the `canUserEditEntityRecord`.

I assume that for sources like site data, we could still rely on similar methods to `canUser` and `canUserEditEntityRecord`.